react-components/stories/apps/admin/AllStats/AllStats.stories.js

```diff
@@ -4,18 +4,38 @@ import appsTableData from "./dataFiles/appsData";
 import distinctLoginData from "./dataFiles/distinctLoginData";
 import jobsData from "./dataFiles/jobsData";
 import usersData from "./dataFiles/usersData";
+import { formatDateObject } from "@cyverse-de/ui-lib/src/util/DateFormatter";
 
-class AllStatsTest extends Component {
-    render() {
-        return (
-            <AllStats
-                appsData={appsTableData.apps}
-                distinctLoginData={distinctLoginData}
-                jobsData={jobsData.jobs}
-                usersData={usersData.users}
-            />
-        );
-    }
+function AllStatsTest(props) {
+    const [selectedStartDate, setSelectedStartDate] = React.useState(
+        new Date()
+    );
+    const [selectedEndDate, setSelectedEndDate] = React.useState(new Date());
+
+    const presenter = {
+        handleStartDateChange: (date) => {
+            let formattedDate = formatDateObject(date, "YYYY-MM-DD");
+            console.log(formattedDate);
+            setSelectedStartDate(formattedDate);
+        },
+        handleEndDateChange: (date) => {
+            let formattedDate = formatDateObject(date, "YYYY-MM-DD");
+            console.log(formattedDate);
+            setSelectedEndDate(formattedDate);
```

Comment:
I think these `setSelectedStartDate` and `setSelectedEndDate` functions should be setting the new `date` object, rather than the `formattedDate` string.

I'm guessing that's what `KeyboardDatePicker` is expecting, and that's also how the original `handleDateChange` function worked in `datePicker`:
```javascript
    const [selectedDate, setSelectedDate] = React.useState(new Date());
    const handleDateChange = (date) => {
        let formattedDate = formatDateObject(date, "YYYY-MM-DD");
        console.log(formattedDate);
        setSelectedDate(date);
    };
```

Reply:
You're right. `formattedDate` was the one that is supposed to be sent to Jack's code and date is the one that `KeyboardDatePicker` receives. I'll make the necessary changes. Thanks for noticing this 👍